client/src/modules/authorization/components/SignUpForm.jsx

```diff
@@ -0,0 +1,309 @@
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const { status } = err.response;
+          const newState = { pending: false };
+
+          if (status === 406) {
+            const { errors } = err.response;
+            newState.higherLevelErrors = errors;
+          }
+
+          this.setState(newState);
+        }
+      });
+  };
+
+  renderSignUpForm = () => {
+    const {
+      higherLevelErrors: {
+        nameError,
+        emailError,
+        passwordConfirmError,
+        passwordError
+      },
+      isFormValid,
+      pending
+    } = this.state;
+    const { onCancel } = this.props;
+
+    return (
+      <Fragment>
+        <h1 className="Sign-Up-Form__heading">
+          <FormattedMessage id="authorization.create-account" />
+        </h1>
+        <form className="Sign-Up-Form__form" noValidate>
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={nameError}
+            focus
+            labelId="authorization.input.username.label"
+            name="name"
+            onChange={this.onNameChange}
+            type="text"
+            validator={this.nameValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={emailError}
+            labelId="authorization.input.email.label"
+            name="email"
+            onChange={this.onEmailChange}
+            type="text"
+            validator={this.emailValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={passwordError}
+            labelId="authorization.input.password.label"
+            name="password"
+            onChange={this.onPasswordChange}
+            type="password"
+            validator={this.passwordValidator}
+          />
+          <SignUpInput
+            disabled={pending}
+            externalErrorId={passwordConfirmError}
+            labelId="authorization.input.password.confirm"
+            name="confirm"
+            onChange={this.onPasswordConfirmChange}
+            type="password"
+          />
+          <div className="Sign-Up-Form__buttons">
+            <button
+              className="primary-button"
+              type="button"
+              disabled={pending}
+              onClick={onCancel}
+            >
+              <FormattedMessage id="common.button.cancel" />
+            </button>
+            <button
+              className="primary-button Sign-Up-Form__confirm"
+              type="button"
+              disabled={pending || !isFormValid}
+              onClick={this.handleSignUp}
+            >
+              <FormattedMessage id="authorization.sign-up" />
+              {pending && (
+                <Preloader
+                  size={PreloaderSize.SMALL}
+                  theme={PreloaderTheme.DARK}
+                />
+              )}
+            </button>
```

Comment:
would be better to use `<PendingButton /> ` here? It has pending preloader included inside.